### PR TITLE
fix/credentials

### PR DESCRIPTION
**Pull request by kovapatrik**
Fixed the async credential fetch.

I've also deleted the error which were thrown when the token/key was undefined at the end of the fetch. Instead it's being handled by the frontend, so the user could see if anything went wrong.

### CODE REVIEW

File: homebridge-ui/server.js [hunk 1]
```diff
@@ -155,9 +158,6 @@ class UiServer extends HomebridgePluginUiServer {
       }
       i++;
     }
-    if (!device.token || !device.key) {
-      throw new Error(`[${device.name}] Authentication failed, token/key undefined.`);
-    }
     this.logger.debug(`Token: ${device.token}, Key: ${device.key}`);
```
Comment (dkerr64) [inline homebridge-ui/server.js:161]: Lets add `[${device.name}]` to this debug statement

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "displayName": "Homebridge Midea",
```
Comment (dkerr64) [inline package.json:2]: While we are in package.json lets update...
`"node": "^18.17.0 || ^20.9.0",`
and I think we should change to this as well...
`"displayName": "Homebridge Midea Platform",`

Comment (dkerr64) [reply package.json:2]: The node version string was accepted into the plugin template by the maintainer of that.  See https://github.com/homebridge/homebridge-plugin-template/issues/62

**Comment by dkerr64** [comment]
Tag issue number #34

**Review by dkerr64 — CHANGES_REQUESTED** [review]

**Comment by kovapatrik** [comment]
Applied the suggestions. Can I merge?

**Comment by kovapatrik** [comment]
Forgot about the name tho. It's getting late...:D

**Review by dkerr64 — APPROVED** [review]

**Pull request merged**